src/librustc/mir/mod.rs

```diff
@@ -1662,6 +1664,13 @@ pub enum StatementKind<'tcx> {
     Nop,
 }
 
+/// The `FakeReadCause` describes the type of pattern why a `FakeRead` statement exists.
+#[derive(Copy, Clone, RustcEncodable, RustcDecodable, Debug)]
+pub enum FakeReadCause {
+    ForMatch,
+    ForLet,
```

Comment:
Let's add some comments and links here.

I think the best option is to move the comments from the point of insertion to here, and then (in those spots) just refer to 

```
// See comments on `FakeReadCause::ForLet
```

This is also good because we may insert fake read from multiple places, but this centralizes the comment in one clearly identifiable location.